plugin/trino-resource-group-managers/src/main/java/io/trino/plugin/resourcegroups/StaticSelector.java

```diff
@@ -41,81 +42,68 @@ public class StaticSelector
-    private final Set<String> variableNames;
+    private final List<SelectionMatcher> selectionMatchers;
 
     public StaticSelector(
             Optional<Pattern> userRegex,
             Optional<Pattern> userGroupRegex,
+            Optional<Pattern> originalUserRegex,
+            Optional<Pattern> authenticatedUserRegex,
             Optional<Pattern> sourceRegex,
             Optional<List<String>> clientTags,
             Optional<SelectorResourceEstimate> selectorResourceEstimate,
             Optional<String> queryType,
             ResourceGroupIdTemplate group)
     {
         this.userRegex = requireNonNull(userRegex, "userRegex is null");
-        this.userGroupRegex = requireNonNull(userGroupRegex, "userGroupRegex is null");
-        this.sourceRegex = requireNonNull(sourceRegex, "sourceRegex is null");
+        requireNonNull(userGroupRegex, "userGroupRegex is null");
+        requireNonNull(originalUserRegex, "originalUserRegex is null");
+        requireNonNull(authenticatedUserRegex, "authenticatedUserRegex is null");
+        requireNonNull(sourceRegex, "sourceRegex is null");
         requireNonNull(clientTags, "clientTags is null");
-        this.clientTags = ImmutableSet.copyOf(clientTags.orElse(ImmutableList.of()));
-        this.selectorResourceEstimate = requireNonNull(selectorResourceEstimate, "selectorResourceEstimate is null");
-        this.queryType = requireNonNull(queryType, "queryType is null");
+        requireNonNull(selectorResourceEstimate, "selectorResourceEstimate is null");
+        requireNonNull(queryType, "queryType is null");
         this.group = requireNonNull(group, "group is null");
 
+        ImmutableList.Builder<SelectionMatcher> selectionMatcherBuilder = ImmutableList.builder();
         HashSet<String> variableNames = new HashSet<>(ImmutableList.of(USER_VARIABLE, SOURCE_VARIABLE));
-        userRegex.ifPresent(u -> addNamedGroups(u, variableNames));
-        sourceRegex.ifPresent(s -> addNamedGroups(s, variableNames));
-        this.variableNames = ImmutableSet.copyOf(variableNames);
-
+        userRegex.ifPresent(u -> {
+            addNamedGroups(u, variableNames);
+            selectionMatcherBuilder.add(new PatternMatcher(variableNames, u, SelectionCriteria::getUser));
+        });
+        originalUserRegex.ifPresent(u -> {
+            addNamedGroups(u, variableNames);
+            selectionMatcherBuilder.add(new PatternMatcher(variableNames, u, SelectionCriteria::getOriginalUser));
+        });
+        authenticatedUserRegex.ifPresent(u -> {
+            addNamedGroups(u, variableNames);
+            selectionMatcherBuilder.add(new OptionalPatternMatcher(variableNames, u, SelectionCriteria::getAuthenticatedUser));
+        });
+        sourceRegex.ifPresent(s -> {
+            addNamedGroups(s, variableNames);
+            selectionMatcherBuilder.add(new OptionalPatternMatcher(variableNames, s, SelectionCriteria::getSource));
+        });
```

Comment:
I think the behavior has changed here. Previously, when `source` was equal to `Optional.empty()` and `sourceRegex` was set to `^$`, we would match the source. Now, we do not match.

Reply:
let's undo this change

Reply:
Ok I restored the original empty-match behavior for `source` and updated `authenticatedUser` to match, for consistency.